### PR TITLE
PLT-4233 When internet reconnects, reconnect websocket and grab latest posts, etc.

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
This needs more testing but generally it should reconnect the websocket and pull latest posts/channels/statuses after the computer reconnects to the internet and the user clicks on the app.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4233

Marked WIP for now, I will ask for testing next week after the release.

### CODE REVIEW

**Comment by jasonblais** [comment]
Tested separately on Windows 10, Chrome and Windows 10, Desktop App:

1. disconnecting from WiFi, then connecting: No issues, messages posted while disconnected show up
2. putting computer in sleep mode, then reopening: No issues, messages posted while disconnected show up
3. disconnecting from Google network tab: messages continue to post in center channel - not sure if this is expected?
   - JS errors are fired notifying that I'm disconnected, which is probably expected, but I can still see the messages showing up

image

@jwilander Can you take a look at number 3? Also let me know if there's anything else I should test

**Comment by jwilander** [comment]
Oh I know what's happened with 3, the Chrome network tab does not disconnect the websocket when you throttle the connection. I remembered manually disconnecting when I was testing this. If the other cases worked fine then I think we're OK to try this out on pre-release.

**Comment by jasonblais** [comment]
Cool, sounds good. There were no issues with the other cases so I'll queue it for dev review

File: webapp/client/web_client.jsx [hunk 1]
```diff
@@ -14,6 +17,7 @@ class WebClientClass extends Client {
     constructor() {
         super();
         this.enableLogErrorsToConsole(true);
+        this.connected = true;
```
Comment (hmhealey) [inline webapp/client/web_client.jsx]: Shouldn't this default to false?

Comment (jwilander) [inline webapp/client/web_client.jsx]: It seems kind of weird but "connected" really just means "has internet connection" and if you download this file then you should be "connected" already, so it starts as that. Also note that this isn't the WebSocket, just the regular web client

Think I should change the variable name and/or add a comment explaining?

Comment (hmhealey) [inline webapp/client/web_client.jsx]: Oh, that makes sense. Maybe just rename it to `internetConnected` or `hasInternetConnection` to make it clearer?

Comment (jwilander) [inline webapp/client/web_client.jsx]: Agreed, I'll make that change

**Comment by jwilander** [comment]
@hmhealey change made

**Pull request merged**